### PR TITLE
Add publish Action for Nextion components

### PR DESCRIPTION
**Pull request by pkejval**
## Description:

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#7646

## Checklist:

  - [X] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/binary_sensor/nextion.rst [hunk 1]
```diff
@@ -88,6 +88,48 @@ should be prefixed with the page name (page0/page1).
 
 ``nextion_component_name: page0.r0``
 
+.. _binary_sensor-nextion-publish_action:
+
+``binary_sensor.nextion.publish`` Action
+------------------------------------------
```
Comment (kbx81) [inline components/binary_sensor/nextion.rst:94]:
```suggestion
----------------------------------------
```

Comment (kbx81) [inline components/binary_sensor/nextion.rst]:
```suggestion
----------------------------------------
```

File: components/binary_sensor/nextion.rst [hunk 2]
```diff
@@ -88,6 +88,48 @@ should be prefixed with the page name (page0/page1).
...
+          send_to_nextion: true
+      # Templated
+      - binary_sensor.nextion.publish:
+          id: nextion_bsensor
+          state: !lambda 'return true;'
+          # These are optional. Defaults to true.
+          publish_state: true
+          send_to_nextion: true
+
+Configuration options:
```
Comment (kbx81) [inline components/binary_sensor/nextion.rst]:
```suggestion
Configuration variables:
```

File: components/sensor/nextion.rst [hunk 3]
```diff
@@ -96,6 +96,48 @@ should be prefixed with the page name (page0/page1 or whatever you have changed
 
 ``component_name: page0.humidity``
 
+.. _sensor-nextion-publish_action:
+
+``sensor.nextion.publish`` Action
+---------------------------------------
```
Comment (kbx81) [inline components/sensor/nextion.rst]:
```suggestion
``sensor.nextion.publish`` Action
---------------------------------
```

File: components/switch/nextion.rst [hunk 4]
```diff
@@ -60,6 +60,48 @@ should be prefixed with the page name (page0/page1 or whatever you have changed
 
 ``component_name: page0.r0``
 
+.. _switch-nextion-publish_action:
+
+``switch.nextion.publish`` Action
+---------------------------------------
```
Comment (kbx81) [inline components/switch/nextion.rst]:
```suggestion
``switch.nextion.publish`` Action
---------------------------------
```

**Review by kbx81 — CHANGES_REQUESTED** [review]
Just a couple RST tweaks -- also, please wrap long lines at no more than 120 columns. Thanks!

File: components/sensor/nextion.rst [hunk 5]
```diff
@@ -96,6 +96,48 @@ should be prefixed with the page name (page0/page1 or whatever you have changed
...
+          send_to_nextion: true
+      # Templated
+      - sensor.nextion.publish:
+          id: nextion_sensor
+          state: !lambda 'return 100.0;'
+          # These are optional. Defaults to true.
+          publish_state: true
+          send_to_nextion: true
+
+Configuration options:
```
Comment (kbx81) [inline components/sensor/nextion.rst]:
```suggestion
Configuration variables:
```

File: components/text_sensor/nextion.rst [hunk 6]
```diff
@@ -56,6 +56,48 @@ should be prefixed with the page name (page0/page1 or whatever you have changed
...
+          send_to_nextion: true
+      # Templated
+      - text_sensor.nextion.publish:
+          id: nextion_text
+          state: !lambda 'return "Hello World";'
+          # These are optional. Defaults to true.
+          publish_state: true
+          send_to_nextion: true
+
+Configuration options:
```
Comment (kbx81) [inline components/text_sensor/nextion.rst]:
```suggestion
Configuration variables:
```

**Review by kbx81 — CHANGES_REQUESTED** [review]
...aaannnddd realized I missed these, too. haha

**Comment by pkejval** [comment]
> Just a couple RST tweaks -- also, please wrap long lines at no more than 120 columns. Thanks!

Why do I need to wrap lines? Original documentation has lines longer than 120 columns. These should be wrapped too? I think this is beyond my PR.

**Comment by kbx81** [comment]
> > Just a couple RST tweaks -- also, please wrap long lines at no more than 120 columns. Thanks!
> 
> Why do I need to wrap lines? Original documentation has lines longer than 120 columns. These should be wrapped too? I think this is beyond my PR.

[Last bullet point here](https://esphome.io/guides/contributing#notes) -- please do this on any new lines you've added to the document. You don't need to go through the whole document and do this, but we also certainly wouldn't mind if you do. 😄

**Review by kbx81 — APPROVED** [review]

**Pull request merged**